.github/workflows/e2e-tests.yaml

```diff
@@ -0,0 +1,43 @@
+name: e2e
+on:
+  pull_request:
+  push:
+    branches:
+      - 'release-*'
+      - 'main'
+    tags:
+      - 'v*'
+env:
+  kind-version: 'v0.11.1'
+  kind-image: 'kindest/node:v1.22.0'
+  golang-version: '1.16'
+
+jobs:
+  e2e-tests:
+    name: E2E tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.golang-version }}
+      - name: Build images
+        run: |
+          export IMAGE_TAG_BASE=monitoring-stack-operator
+          make operator-image
+      - name: Start KinD
```

Comment:
May in a separate pr. I would be nice to have this as a `hack/kind-setup.sh` script.

Reply:
I `kind` of agree with you but to have a default imho is fine, otherwise we could document it but why document code :D